### PR TITLE
Extend api

### PR DESCRIPTION
**Pull request by csbnw**
**Description**

Extend the API so that cudawrappers can also be used in [PMT](https://git.astron.nl/RD/pmt):

- Added `cu::Device::getUUID()`
- Added `cudawrappers::nvml` target with only the functionality that PMT needs

**Related issues**:

- N.A.

**Instructions to review the pull request**

- Check that CHANGELOG.md has been updated if necessary

### CODE REVIEW

File: include/cudawrappers/nvml.hpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+inline void checkNvmlCall(nvmlReturn_t result) {
+  if (result != NVML_SUCCESS) throw Error(result);
+}
+
+class Device {
+ public:
+  Device(int index) {
+    checkNvmlCall(nvmlInit());
```
Comment (matmanc) [inline include/cudawrappers/nvml.hpp:30]: what happens if you init multiple times by creating 2 devices?

File: include/cudawrappers/nvml.hpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    checkNvmlCall(nvmlDeviceGetHandleByIndex(index, &device_));
+  }
+
+  Device(cu::Device& device) {
+    checkNvmlCall(nvmlInit());
```
Comment (matmanc) [inline include/cudawrappers/nvml.hpp:35]: Same here

File: include/cudawrappers/nvml.hpp (hunk 1 above)
Comment (csbnw) [reply include/cudawrappers/nvml.hpp]: After giving it a quick test, calling `nvmlInit` multiple times doesn't seem to cause any problems.

**Comment by csbnw** [comment]
After a discussion with @matmanc, we concluded that it is potentially dangerous if the `nvml::Device` initializes NVML and especially destroys it again (when other instances may still be active). A new `nvml::Context` class is added to guard against this. A `nvml::Context` must now be created first and next passed to the `nvml::Device` constructor. We also discussed adding `Context::getDevice`, but I think the current solution is cleaner.

**Review by matmanc — APPROVED** [review]
looks good

**Pull request merged**